crates/sui-core/src/traffic_controller/mod.rs

```diff
@@ -206,40 +215,60 @@ async fn run_tally_loop(
-                        &mut spam_policy,
-                        &policy_config,
-                        &node_fw_client,
-                        &fw_config,
-                        tally.clone(),
-                        spam_blocklists.clone(),
-                        metrics.clone(),
-                    )
-                    .await {
-                        warn!("Error handling spam tally: {}", err);
-                    }
+            received = receiver.recv() => {
+                metrics.tallies.inc();
+                match received {
+                    Some(tally) => {
+                        if let Err(err) = handle_spam_tally(
+                            &mut spam_policy,
+                            &policy_config,
+                            &node_fw_client,
+                            &fw_config,
+                            tally.clone(),
+                            spam_blocklists.clone(),
+                            metrics.clone(),
+                        )
+                        .await {
+                            warn!("Error handling spam tally: {}", err);
+                        }
 
-                    if let Err(err) = handle_error_tally(
-                        &mut error_policy,
-                        &policy_config,
-                        &node_fw_client,
-                        &fw_config,
-                        tally,
-                        error_blocklists.clone(),
-                        metrics.clone(),
-                    )
-                    .await {
-                        warn!("Error handling error tally: {}", err);
+                        if let Err(err) = handle_error_tally(
+                            &mut error_policy,
+                            &policy_config,
+                            &node_fw_client,
+                            &fw_config,
+                            tally,
+                            error_blocklists.clone(),
+                            metrics.clone(),
+                        )
+                        .await {
+                            warn!("Error handling error tally: {}", err);
+                        }
+                    }
+                    None => {
+                        info!("TrafficController tally channel closed by all senders");
+                        return;
+                    }
+                }
+            }
+            // Dead man's switch - if we suspect something is sinking all traffic to node, disable nodefw
+            _ = tokio::time::sleep(tokio::time::Duration::from_secs(timeout)) => {
+                if let Some(fw_config) = &fw_config {
+                    if fw_config.delegate_spam_blocking || fw_config.delegate_error_blocking {
+                        error!("No traffic tallies received in {} seconds! Disabling BPF.", fw_config.killswitch_timeout);
+                        let killswitch_path = fw_config.killswitch_path.join("__KILLSWITCH__");
```

Comment:
if the user sets `killswitch_path` to a directory that doesn't exist (or they don't have write permissions to) this will fail, but they won't know until it is too late. can you write a file to this dir at start up and panic if it fails?

Reply:
actually - perhaps we should create the `__KILLSWITCH__` put `off` or something in it, and then overwrite it with `on` when it trips. This way the system will work even if the disk is full

Reply:
smart